### ISSUE TITLE
Support AST::Table#map_column!

### ISSUE THREAD

**Issue by senny** [cucumber-attic/cuke4duke#60]
It would be great if there was a possibility to manipulate column data before using Table#diffHashes().

**Comment by aslakhellesoy** [comment]
I see you have implemented this. Can you add some code that illustrates how to use it here in the tracker (Use HTML pre tags or a gist)? Something I can put in the wiki when I merge this in?

**Comment by senny** [comment]
Here is an example usage of the comitted code:

TableConverter:

```
public class TableTranslator implements TableConverter{

  public String convert(String key) {
    return TranslatorUtil.translate(key);
  }

}
```

Step Definition:

```
public void translateTable(cuke4duke.Table table) {
  table.convertColumn("key", new TableTranslator());
}
```

**Comment by aslakhellesoy** [comment]
If I understand correctly, convertColumn maps to map_column! and convertHeaders to map_headers!. Why did you come up with new method names?

Another thing that puzzles me is the use of convertColumn. If you read the RDoc for map_column!, it says: "Change how #hashes converts column values. The column_name argument identifies the column and conversion_proc performs the conversion for each cell in that column.". With that in mind, it confuses me to have an argument named "key". That makes me think of a header cell, but in fact we're converting cells on each non-header row in a column. Maybe rename "key" to "cellValue"?

That leads me to the next question. An interface with one method: String convert(String cellValue) - is that a TableConverter? Or a CellConverter? To me it sounds like the latter.

Let me know what you think about this. Names are important.

**Comment by senny** [comment]
I think all of your points are valid improvements over my code. My implementation is more of a short hack to get it working in our environment than a tought-trough addition to cuke4duke. Of course I can change the names accordingly if you want.

**Comment by aslakhellesoy** [comment]
Ca marche. I can do the appropriate changes after merging in your code. Just wanted to check with you first :-)

**Comment by aslakhellesoy** [comment]
Attribution. Closed by a3c93526fb5fd798f0dcf2a8cd086d5395124295.
